src/NUnitFramework/tests/Constraints/PropertyTests.cs

```diff
@@ -206,4 +208,68 @@ public void FailureMessageContainsChainedConstraintMessage()
             Assert.IsTrue(textMessageWriter.ToString().Contains("Extra"));
         }
     }
+
+    [TestFixture(TestOf = typeof(PropertyConstraint))]
+    public class PropertyHierarchyTests
+    {
+        [Test]
+        public void PropertyExists_PropertyDefinedInDerivedClass_ShouldExist()
+        {
+            var sut = new PropertyConstraint(nameof(Derived.SomeProperty), new EqualConstraint(42));
+
+            var instance = (Base)new Derived();
+            var actual = sut.ApplyTo(instance);
+
+            Assert.That(actual, Has.Property(nameof(ConstraintResult.Status)).EqualTo(ConstraintStatus.Success));
+        }
+
+        class Base { }
+        class Derived : Base
+        {
+            public int SomeProperty { get; set; } = 42;
+        }
+
+
+        private int[] _array;
+        private PropertyConstraint _countPropertyConstraint;
+
+        [SetUp]
+        public void BeforeEveryTest()
+        {
+            _array = new[] { 1, 2, 3 };
+            _countPropertyConstraint = new PropertyConstraint(nameof(IList<object>.Count), new EqualConstraint(_array.Length));
+        }
+
+        [Test]
+        public void PropertyExists_PropertyDefinedInInterface_ShouldNotExist_WhenNotExplicitlyDefined()
```

Comment:
The word 'explicitly' could be confusing to future readers because arrays explicitly rather than implicitly implement the Count property. What do you think about:
```suggestion
        public void ExplicitlyImplementedProperty_ShouldNotExist_ViaImplementingType()
```

Reply:
(We don't usually use the same prefix on a bunch of tests like `PropertyExists_`. Do you feel we would lose too much if we don't use it?)

Reply:
I'm ok with suggested name. You're right, "PropertyExists_" might have been an exaggeration